src/Core/Label.js

```diff
@@ -130,15 +133,33 @@ class Label extends THREE.Object3D {
             this.boundaries.right = x + this.offset.right + this.padding;
             this.boundaries.top = y + this.offset.top - this.padding;
             this.boundaries.bottom = y + this.offset.bottom + this.padding;
+
+            // The boundaries of the label are the union of the boundaries of the text
+            // and the boundaries of the icon, if it exists.
+            // Checking if this.icon is not only zeros is mandatory, to prevent case
+            // when a boundary is set to x or y coordinate
+            if (JSON.stringify(this.iconOffset) !== JSON.stringify({ left: 0, right: 0, top: 0, bottom: 0 })) {
```

Comment:
could you check properties?

Reply:
I now check for all `left`, `right`, `top` and `bottom` properties.